packages/editor/src/components/blocks/radio/Radio.tsx

```diff
@@ -0,0 +1,88 @@
+import type { Prettify, Radio, AlignmentType } from '@axonivy/form-editor-protocol';
+import type { ComponentConfig, FieldOption, UiComponentProps } from '../../../types/config';
+import './Radio.css';
+import { baseComponentFields, defaultBaseComponent } from '../base';
+import IconSvg from './Radio.svg?react';
+import { Field, Label, RadioGroup, RadioGroupItem } from '@axonivy/ui-components';
+
+type RadioProps = Prettify<Radio>;
+
+const layoutOptions: FieldOption<AlignmentType>[] = [
+  { label: 'horizontal', value: 'lineDirection' },
+  { label: 'vertical', value: 'pageDirection' }
+] as const;
+
+export const defaultInputProps: Radio = {
+  label: 'Label',
+  layout: 'lineDirection',
+  value: '',
+  staticItems: [
+    { label: 'Option 1', value: 'Option 1' },
+    { label: 'Option 2', value: 'Option 2' }
+  ],
+  dynamicItemsList: '',
+  dynamicItemsLabel: '',
+  dynamicItemsValue: '',
+  ...defaultBaseComponent
+} as const;
+
+export const RadioComponent: ComponentConfig<RadioProps> = {
+  name: 'Radio',
+  category: 'Elements',
+  subcategory: 'Interactions',
+  icon: <IconSvg />,
+  description: 'A simple input with a label',
+  defaultProps: defaultInputProps,
+  render: props => <UiBlock {...props} />,
+  create: ({ label, value, ...defaultProps }) => ({ ...defaultInputProps, label, value, ...defaultProps }),
+  outlineInfo: component => component.label,
+  fields: {
+    label: { subsection: 'General', label: 'Label', type: 'text' },
+    value: { subsection: 'General', label: 'Value', type: 'textBrowser' },
+    layout: { subsection: 'General', label: 'Orientation', type: 'select', options: layoutOptions },
+    staticItems: { subsection: 'Static Options', label: '', type: 'selectTable' },
+    dynamicItemsList: {
+      subsection: 'Dynamic Options',
+      label: 'List of Objects',
+      type: 'textBrowser',
+      options: { onlyListTypes: true, placeholder: 'e.g. #{data.dynamicList}' }
+    },
+    dynamicItemsLabel: {
+      subsection: 'Dynamic Options',
+      label: 'Object Label',
+      type: 'textBrowser',
+      options: {
+        onlyAttributes: true,
+        placeholder: 'Enter attribute (or leave blank to select entire object)'
+      },
+      hide: data => data.dynamicItemsList.length == 0
+    },
+    dynamicItemsValue: {
+      subsection: 'Dynamic Options',
+      label: 'Object Value',
+      type: 'textBrowser',
+      options: {
+        onlyAttributes: true,
+        placeholder: 'Enter attribute (or leave blank to select entire object)'
+      },
+      hide: data => data.dynamicItemsList.length == 0
+    },
```

Comment:
can this also be generallized with the stuff from the select component?

Reply:
Yes, I can try 👍